app/Http/Controllers/TarnibController.php

```diff
@@ -1,133 +1,182 @@
+                'player_2_cards' => json_encode($this->sortCards(array_slice($random_cards, 13, 13))),
+                'player_3_cards' => json_encode($this->sortCards(array_slice($random_cards, 26, 13))),
+                'player_4_cards' => json_encode($this->sortCards(array_slice($random_cards, 39, 13))),
+                'turn' => 1,
+                'team_1_score' => 0,
+                'team_2_score' => 0,
+                'dealer' => $dealer,
+                'bids_data' => json_encode([
+                    'bids' => [0,0,0,0],
+                    'current_bidder' => ($round_number+1)%4
+                ]),
+                'current_play' => json_encode(['','','','']),
+                'player_turn' => -1,
+
+            ]);
+
+            return $round_id;
+        }
 
-        //$round_count = count(DB::table('rounds')->where('id', $room->round_id)->get());
+        function joinRoom(Request $request){
+            $room = DB::table('rooms')->where('room_id', $request->room_id)->first();
+            $player_seat = '';
+            if($room->player_1 == $request->player_name) $player_seat = 1;
+            else if($room->player_2 == $request->player_name) $player_seat = 2;
+            else if($room->player_3 == $request->player_name) $player_seat = 3;
+            else if($room->player_4 == $request->player_name) $player_seat = 4;
+
+
+            $round_number = count(DB::table('rounds')->where('room_id', $room_id)->get());
+
+            //$round_count = count(DB::table('rounds')->where('id', $room->round_id)->get());
+
+            $round = DB::table('rounds')->where('id', $room->round_id)->first();
+            if($player_seat != ''){
+                if($player_seat == 1){
+                    $cards = json_decode($round->player_1_cards);
+                }else if($player_seat == 2){
+                    $cards = json_decode($round->player_2_cards);
+                }else if($player_seat == 3){
+                    $cards = json_decode($round->player_3_cards);
+                }else if($player_seat == 4){
+                    $cards = json_decode($round->player_4_cards);
+                }else{
+                    $cards = [];
+                }
+            }
+
+            // if($cards != []){
+            //     $cards = $this->sortCards($cards);
+            // }
+
+            
+
+            return view('game')->with([
+                'player_name' => $request->player_name,
+                'player_seat' => $player_seat,
+                'room_id' => $room->room_id,
+                'player_1' => $room->player_1,
+                'player_2' => $room->player_2,
+                'player_3' => $room->player_3,
+                'player_4' => $room->player_4,
+                'round' => $round_number,
+                'turn' => $round->turn,
+                'cards' => $cards,
+                'bids_data' => json_decode($round->bids_data),
```

Comment:
its a bit confusing to know when developing which attributes already were json decoded and which are not